device-registry/src/main/scala/org/genivi/sota/device_registry/PublicCredentialsResource.scala

```diff
@@ -0,0 +1,82 @@
+/**
+  * Copyright: Copyright (C) 2016, ATS Advanced Telematic Systems GmbH
+  * License: MPL-2.0
+  */
+package org.genivi.sota.device_registry
+
+import akka.http.scaladsl.marshalling.Marshaller._
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.{Directive1, Route}
+import akka.http.scaladsl.util.FastFuture
+import akka.stream.ActorMaterializer
+import io.circe.generic.auto._
+import java.util.Base64
+import org.genivi.sota.data.{DeviceT, Namespace, Uuid}
+import org.genivi.sota.device_registry.db.{PublicCredentialsRepository, DeviceRepository}
+import org.genivi.sota.device_registry.common.Errors
+import org.genivi.sota.http.{AuthedNamespaceScope, Scopes}
+import org.genivi.sota.marshalling.CirceMarshallingSupport._
+import org.genivi.sota.messaging.MessageBusPublisher
+import org.genivi.sota.messaging.Messages.{DeviceCreated, DevicePublicCredentialsSet}
+import slick.driver.MySQLDriver.api._
+import scala.concurrent.{ExecutionContext, Future}
+
+object PublicCredentialsResource {
+  final case class FetchPublicCredentials(uuid: Uuid, credentials: String)
+}
+
+class PublicCredentialsResource(authNamespace: Directive1[AuthedNamespaceScope],
+                          messageBus: MessageBusPublisher,
+                          deviceNamespaceAuthorizer: Directive1[Uuid])
+                         (implicit db: Database,
+                          mat: ActorMaterializer,
+                          ec: ExecutionContext) {
+  import PublicCredentialsResource._
+  lazy val base64Decoder = Base64.getDecoder()
+  lazy val base64Encoder = Base64.getEncoder()
+
+  def fetchPublicCredentials(uuid: Uuid): Route =
+    complete(db.run(PublicCredentialsRepository.findByUuid(uuid)).map { creds =>
+               FetchPublicCredentials(uuid, base64Encoder.encodeToString(creds))
+             })
+
+  def createDeviceWithPublicCredentials(ns: Namespace, devT: DeviceT): Route = {
+    val act = (devT.deviceId, devT.credentials) match {
+      case (Some(devId), Some(base64Credentials)) => {
+        val credentials = base64Decoder.decode(base64Credentials)
```

Comment:
I would not decode credentials here, just take bytes of the string. For instance, there are textual formats of certificates, which are not Base64, and additional encoding will make no sense.